runtime/autoload/dist/ft.vim

```diff
@@ -286,6 +286,39 @@ export def FTe()
   endif
 enddef
 
+def IsForth(): bool
+  var n = 1
+  while n < 100 && n <= line("$")
+    # Forth comments and colon definitions
+    if getline(n) =~ '^[:(\\] '
+      return true
+    endif
+    n += 1
+  endwhile
+  return false
+enddef
+
+# Distinguish between Forth and Fortran
+export def FTf()
+  if exists("g:filetype_f")
+    exe "setf " .. g:filetype_f
+    return
+  endif
+
+  var first_line = getline(nextnonblank(1))
+
+  # SwiftForth block comment, line is usually filled with '-' or '='
+  var header_comment   = first_line =~ '^{\%(\s\|$\)'
+  # SwiftForth OPTIONAL, sometimes appears before the header comment
+  var optional_include = first_line =~ '^OPTIONAL\s'
```

Comment:
Could the SwiftForth checks give false positives for F#? If not, I think they would be better moved to the IsForth() function.